File: perl/DayX/DayX.pl
#!/usr/bin/env perl
# DayX.pl
#
# First attempt at doing Day X of Advent of Code 2025 with perl.

use warnings;
use strict;

# Setup an empty array
my @array;
my @array_output;
# Open the file, remove \n and then put into array
open( my $fh, "<", "DayX_input.txt")
#open(my $fh, "<", "DayX_test_input.txt")
    or die "Failed to open file : $!\n";
while(<$fh>) {
    chomp;
    push @array, $_;
}
close $fh;

#print join " ", @array;

my $count = 0;
foreach (@array) {
#    print $_;
    if ($_ eq "") {
        $count += 1;
        $array_output[$count] = 0;
    } elsif ($_ eq "\n"){
        print "pass";
    } else {
#        print $count;
        $array_output[$count] += $_;
    }
}
# print join " ", @array_output;

my @sorted_array = sort { $a <=> $b } @array_output;
print "max: $sorted_array[-1]\n";

print "max3: $sorted_array[-1] $sorted_array[-2] $sorted_array[-3]\n";
my $val = $sorted_array[-1] + $sorted_array[-2] + $sorted_array[-3];
print "total: $val\n";

File: python/DayX/DayX.py
# DayX.py
#
# First attempt at doing Day X of Advent of Code 2025

import time
import numpy as np


def read_input(file_name):
    with open(file_name, 'r') as f:
        read_list = f.read().splitlines()

    final_count = [0]
    counter = 0
    for entry in read_list:
        if entry != '':
            final_count[counter] += int(entry)
        else:
            counter += 1
            final_count.append(0)

    final_count = np.asarray(final_count)

    return final_count


def return_max(input_list):
    max_cal = np.max(input_list)
    where = np.where(input_list == max_cal)[0][0]

    return [int(max_cal), where]


def return_top3(input_list):
    sorted_list = sorted(input_list, key=lambda x: -x)
    cal_sum = sum(sorted_list[0:3])
    where_three = np.where(input_list > sorted_list[3])[0]

    return [int(cal_sum), where_three]


def main():
    ## Part 1
    a = read_input("DayX_test_input.txt")
    # a = read_input("DayX_input.txt")
    answer = return_max(a)
    print(f'Elf {answer[1]+1} has {answer[0]} calories worth of food.')


def main2():
    # Part 2
    b = read_input("DayX_test_input.txt")
    # b = read_input("DayX_input.txt")
    answer2 = return_top3(b)
    print(f'Elves {answer2[1] + 1} have {answer2[0]} calories worth of food.')


if __name__ == "__main__":
    ## Part 1
    startTime = time.perf_counter()
    main()
    print(f'{time.perf_counter() - startTime} sec.')

    ## Part 2
    # startTime = time.perf_counter()
    # main2()
    # print(f'{time.perf_counter() - startTime} sec.')
